adi/pqmon.py

```diff
@@ -0,0 +1,547 @@
+
+        @property
+        def msv_carrier_frequency(self):
+            return self._get_iio_dev_attr("msv_carrier_frequency")
+
+        @msv_carrier_frequency.setter
+        def msv_carrier_frequency(self, value):
+            self._set_iio_dev_attr_str("msv_carrier_frequency", value)
+
+        @property
+        def msv_record_length(self):
+            return self._get_iio_dev_attr("msv_record_length")
+
+        @msv_record_length.setter
+        def msv_record_length(self, value):
+            self._set_iio_dev_attr_str("msv_record_length", value)
+
+        @property
+        def msv_threshold(self):
+            return self._get_iio_dev_attr("msv_threshold")
+
+        @msv_threshold.setter
+        def msv_threshold(self, value):
+            self._set_iio_dev_attr_str("msv_threshold", value)
+
+        @property
+        def sampling_frequency(self):
+            return self._get_iio_dev_attr("sampling_frequency")
+
+        @property
+        def v_consel(self):
+            return self._get_iio_dev_attr("v_consel")
+
+        @v_consel.setter
+        def v_consel(self, value):
+            if value not in self.v_consel_available:
+                raise ValueError("v_consel must be in v_consel_available")
+            self._set_iio_dev_attr_str("v_consel", value)
+
+        @property
+        def flicker_model(self):
+            return self._get_iio_dev_attr("flicker_model")
+
+        @flicker_model.setter
+        def flicker_model(self, value):
+            if value not in self.flicker_model_available:
+                raise ValueError("flicker_model must be in flicker_model_available")
+            self._set_iio_dev_attr_str("flicker_model", value)
+
+        @property
+        def nominal_frequency(self):
+            return self._get_iio_dev_attr("nominal_frequency")
+
+        @nominal_frequency.setter
+        def nominal_frequency(self, value):
+            if value not in self.nominal_frequency_available:
+                raise ValueError(
+                    "nominal_frequency must be in nominal_frequency_available"
+                )
+            self._set_iio_dev_attr_str("nominal_frequency", str(value))
+
+        @property
+        def firmware_version(self):
+            return self._get_iio_dev_attr("fw_version")
+
+        @property
+        def process_data(self):
+            return self._get_iio_dev_attr("process_data")
+
+        @process_data.setter
+        def process_data(self, value):
+            self._set_iio_dev_attr_str("process_data", value)
+
+        pass
+
+        @property
+        def single_shot_acquisition(self):
+            return self._single_shot_acquisition
+
+        @single_shot_acquisition.setter
+        def single_shot_acquisition(self, value):
+            self._single_shot_acquisition = value
+            if value is True:
+                self.process_data = True
+
+    def rx(self):
+        self._rx_channel_names = [
```

Comment:
What is the reasoning behind a custom rx method? Please document why this is here and why it fixes certain parameters